sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -557,12 +586,63 @@ export class EventHubReceiver extends LinkEntity {
     );
   }
 
+  private async _drainAndAddCredits(
+    onMessage: OnMessage,
+    abortSignal?: AbortSignalLike
+  ): Promise<void> {
+    await delay(0);
+    this._isDrainingQueue = true;
+    while (this.internalQueue.length) {
+      if (!this._hasActiveListeners()) {
+        this._isDrainingQueue = false;
+        return;
+      }
+
+      if (abortSignal && abortSignal.aborted) {
+        this._isDrainingQueue = false;
+        return;
+      }
+
+      if (this._userDefinedOnMessage !== onMessage) {
```

Comment:
When do we expect the 2 to not be the same?

Reply:
This can happen when using the `getEventIterator` method. The reason being that when using the event iterator, it's possible for both `clearHandlers` and `registerHandlers` to be called in the same event loop tick. 
This causes the `if (!this._onMessage)` check in the while loop to fail because a new `_onMessage` handler has been set, but _drainAndAddCredits is still running from the 1st time `registerHandlers` was called.

Reply:
Perhaps, add a comment?